### ISSUE TITLE
[8.0] Build failure in linux.riscv64.Checked - No usable version of clang found

### ISSUE THREAD

**Issue by carlossanlop** [dotnet/runtime#100564]
Affecting both coreclr and mono.

## Build Information

Build: https://dev.azure.com/dnceng-public/public/_build/results?buildId=627418
Build error leg or test failing:
  - CoreCLR: `CoreCLR Product Build linux riscv64 checked` - Step: `Build CoreCLR Runtime`
  - Mono: `linux-riscv64 Release AllSubsets_Mono` - Step: `Build product`

## Error Message

```json
{
  "ErrorPattern": "riscv64.+No usable version of clang found",
  "BuildRetry" : false,
  "ExcludeConsoleLog" : false
}
```

- PR: https://github.com/dotnet/runtime/pull/100547

## CoreCLR

- Job result: https://dev.azure.com/dnceng-public/public/_build/results?buildId=627418&view=logs&j=61a6c216-902b-58a9-7a66-2fc8440918c9&t=e3de8ff2-0932-5298-3d8c-422e717b2560&l=23
- Output:
```
========================== Starting Command Output ===========================
/usr/bin/bash --noprofile --norc /__w/_temp/c9129468-7f8f-4842-8ef9-611f801c422c.sh
Commencing CoreCLR Repo build
__OutputRid: linux-riscv64
Setting up directories for build
Checking prerequisites...
Commencing build of "install" target in "CoreCLR component" for linux.riscv64.Checked in /__w/1/s/artifacts/obj/coreclr/linux.riscv64.Checked
Invoking "/__w/1/s/eng/native/gen-buildsys.sh" "/__w/1/s/src/coreclr" "/__w/1/s/artifacts/obj/coreclr/linux.riscv64.Checked" riscv64 linux clang Checked ""  -DCLR_CMAKE_PGO_INSTRUMENT=0 -DCLR_CMAKE_OPTDATA_PATH= -DCLR_CMAKE_PGO_OPTIMIZE=0 -DCLI_CMAKE_FALLBACK_OS="linux" -DFEATURE_DISTRO_AGNOSTIC_SSL=1  -DCLR_CMAKE_KEEP_NATIVE_SYMBOLS=true
No usable version of clang found.
##[error]Failed to generate "CoreCLR component" build project!

##[error]Bash exited with code '1'.
Finishing: Build CoreCLR Runtime
```

## Mono

- Job result: https://dev.azure.com/dnceng-public/public/_build/results?buildId=627418&view=logs&j=c0070a10-b820-5223-30c5-f4f0fcd26ab8&t=f0688afd-02b5-5c64-6068-1bbaeb6a1e4e
- Output:
```
  Microsoft.NET.Runtime.MonoTargets.Sdk -> /__w/1/s/artifacts/packages/Release/Shipping/Microsoft.NET.Runtime.MonoTargets.Sdk.8.0.5.nupkg
  Running 'sh -c 'build_arch="riscv64" compiler="clang" . "/__w/1/s/eng/common/native/init-compiler.sh" && TARGET_BUILD_ARCH=riscv64 PKG_CONFIG_PATH=/crossrootfs/riscv64/usr/lib/riscv64-linux-gnu/pkgconfig cmake -DCMAKE_INSTALL_PREFIX="/__w/1/s/artifacts/obj/mono/linux.riscv64.Release/out" -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_BUILD_TYPE=Release -DGC_SUSPEND=hybrid -DMONO_LIB_NAME=monosgen-2.0 -DMONO_SHARED_LIB_NAME=coreclr -DCMAKE_TOOLCHAIN_FILE=/__w/1/s/eng/common/cross/toolchain.cmake -DSTATIC_COMPONENTS=1 -DMONO_COMPONENTS_RID=linux-riscv64 -DCLR_CMAKE_HOST_ARCH=riscv64 -DCMAKE_C_FLAGS=" -Wl,--build-id=sha1" -DCMAKE_CXX_FLAGS=" -Wl,--build-id=sha1"  "/__w/1/s/src/mono"'' in '/__w/1/s/artifacts/obj/mono/linux.riscv64.Release/'
  No usable version of clang found.
/__w/1/s/src/mono/mono.proj(637,5): error MSB3073: The command "sh -c 'build_arch="riscv64" compiler="clang" . "/__w/1/s/eng/common/native/init-compiler.sh" && TARGET_BUILD_ARCH=riscv64 PKG_CONFIG_PATH=/crossrootfs/riscv64/usr/lib/riscv64-linux-gnu/pkgconfig cmake -DCMAKE_INSTALL_PREFIX="/__w/1/s/artifacts/obj/mono/linux.riscv64.Release/out" -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_BUILD_TYPE=Release -DGC_SUSPEND=hybrid -DMONO_LIB_NAME=monosgen-2.0 -DMONO_SHARED_LIB_NAME=coreclr -DCMAKE_TOOLCHAIN_FILE=/__w/1/s/eng/common/cross/toolchain.cmake -DSTATIC_COMPONENTS=1 -DMONO_COMPONENTS_RID=linux-riscv64 -DCLR_CMAKE_HOST_ARCH=riscv64 -DCMAKE_C_FLAGS=" -Wl,--build-id=sha1" -DCMAKE_CXX_FLAGS=" -Wl,--build-id=sha1"  "/__w/1/s/src/mono"'" exited with code 1.
##[error]src/mono/mono.proj(637,5): error MSB3073: (NETCORE_ENGINEERING_TELEMETRY=Build) The command "sh -c 'build_arch="riscv64" compiler="clang" . "/__w/1/s/eng/common/native/init-compiler.sh" && TARGET_BUILD_ARCH=riscv64 PKG_CONFIG_PATH=/crossrootfs/riscv64/usr/lib/riscv64-linux-gnu/pkgconfig cmake -DCMAKE_INSTALL_PREFIX="/__w/1/s/artifacts/obj/mono/linux.riscv64.Release/out" -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_BUILD_TYPE=Release -DGC_SUSPEND=hybrid -DMONO_LIB_NAME=monosgen-2.0 -DMONO_SHARED_LIB_NAME=coreclr -DCMAKE_TOOLCHAIN_FILE=/__w/1/s/eng/common/cross/toolchain.cmake -DSTATIC_COMPONENTS=1 -DMONO_COMPONENTS_RID=linux-riscv64 -DCLR_CMAKE_HOST_ARCH=riscv64 -DCMAKE_C_FLAGS=" -Wl,--build-id=sha1" -DCMAKE_CXX_FLAGS=" -Wl,--build-id=sha1"  "/__w/1/s/src/mono"'" exited with code 1.

Build FAILED.

/__w/1/s/src/mono/mono.proj(637,5): error MSB3073: The command "sh -c 'build_arch="riscv64" compiler="clang" . "/__w/1/s/eng/common/native/init-compiler.sh" && TARGET_BUILD_ARCH=riscv64 PKG_CONFIG_PATH=/crossrootfs/riscv64/usr/lib/riscv64-linux-gnu/pkgconfig cmake -DCMAKE_INSTALL_PREFIX="/__w/1/s/artifacts/obj/mono/linux.riscv64.Release/out" -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_BUILD_TYPE=Release -DGC_SUSPEND=hybrid -DMONO_LIB_NAME=monosgen-2.0 -DMONO_SHARED_LIB_NAME=coreclr -DCMAKE_TOOLCHAIN_FILE=/__w/1/s/eng/common/cross/toolchain.cmake -DSTATIC_COMPONENTS=1 -DMONO_COMPONENTS_RID=linux-riscv64 -DCLR_CMAKE_HOST_ARCH=riscv64 -DCMAKE_C_FLAGS=" -Wl,--build-id=sha1" -DCMAKE_CXX_FLAGS=" -Wl,--build-id=sha1"  "/__w/1/s/src/mono"'" exited with code 1.
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:01:37.17
Build failed with exit code 1. Check errors above.
```

 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=627418
**Error message validated:** `[riscv64.+No usable version of clang found`]
**Result validation:** :x: Known issue did not match with the provided build.
**Validation performed at:** 4/3/2024 12:04:03 AM UTC

#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|

**Comment by jkotas** [comment]
Duplicate of #100215. Fixed by #100258.

@carlossanlop This should be closed after the next merge of release/8.0-staging to release/8.0

**Comment by agocke** [comment]
Closing as dup
